src/Foundatio.Redis/Cache/RedisCacheClient.cs

```diff
@@ -88,14 +87,43 @@ await foreach (var key in server.KeysAsync().ConfigureAwait(false))
         public async Task<int> RemoveByPrefixAsync(string prefix) {
             await LoadScriptsAsync().AnyContext();
 
+            const int chunkSize = 2500;
+            string regex = prefix + "*";
             try {
-                var result = await Database.ScriptEvaluateAsync(_removeByPrefix, new { keys = prefix + "*" }).AnyContext();
-                return (int)result;
+                int total = 0;
+                int index = 0;
+
+                (int cursor, string[] keys) = await ScanKeysAsync(regex, index, chunkSize).AnyContext();
+
+                while (keys.Length != 0 || index < chunkSize) {
+                    total += await RemoveAllAsync(keys).AnyContext();
+                    index += chunkSize;
+                    (cursor, keys) = await ScanKeysAsync(regex, cursor, chunkSize).AnyContext();
+                }
+
+                total += await RemoveAllAsync(keys).AnyContext();
+
+                return total;
             } catch (RedisServerException) {
                 return 0;
             }
         }
 
+        /// <summary>
+        /// Scan for keys matching the prefix
+        /// </summary>
+        /// <remarks>SCAN, SSCAN, HSCAN and ZSCAN return a two elements multi-bulk reply, where the first element
+        /// is a string representing an unsigned 64 bit number (the cursor), and the second element is a multi-bulk
+        /// with an array of elements.</remarks>
+        private async Task<(int, string[])> ScanKeysAsync(string prefix, int index, int chunkSize)
+        {
+            var result = await Database.ExecuteAsync("scan", index, "match", prefix, "count", chunkSize).AnyContext();
+            var uniqueValue = result.ToDictionary().FirstOrDefault();
```

Comment:
What is the shape of the result here? If it's two elements, can we speed this up by selecting the array element?

Reply:
Good idea.
I've made the change: I'm casting directly to the expected types.